api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/SyntheticBeanBuilder.java

```diff
@@ -283,7 +305,20 @@
      * @return this {@code SyntheticBeanBuilder}
      */
     SyntheticBeanBuilder<T> withParam(String key, Class<?> value);
-    // TODO add a variant that takes a `ClassInfo`? the value would be `Class` at runtime
+
+    /**
+     * Adds a {@code Class}-valued parameter to the map of construction/destruction parameters.
+     * The parameter map is passed to the {@linkplain SyntheticBeanCreator creation} and
+     * {@linkplain SyntheticBeanDisposer destruction} function when the bean is instantiated/disposed.
+     * <p>
+     * When looked up from the parameter map in the creation/destruction function, the value will be
+     * an instance of {@link Class}, <em>not</em> a {@code ClassInfo}.
+     *
+     * @param key the parameter key, must not be {@code null}
+     * @param value the parameter value
+     * @return this {@code SyntheticBeanBuilder}
+     */
+    SyntheticBeanBuilder<T> withParam(String key, ClassInfo value);
```

Comment:
I recall you asked for this @graemerocher? And even if not, I thought that makes sense.